src/Types.js

```diff
@@ -3,27 +3,42 @@ export const copyValue = src => src;
 export const cloneValue = src => src;
 
 export const copyArray = (src, dest) => {
-  const srcArray = src;
-  const destArray = dest;
+  if (!src) {
+    return src;
+  }
+
+  if (!dest) {
+    return src.slice();
+  }
 
-  destArray.length = 0;
+  dest.length = 0;
 
-  for (let i = 0; i < srcArray.length; i++) {
-    destArray.push(srcArray[i]);
+  for (let i = 0; i < src.length; i++) {
+    dest.push(dest[i]);
   }
 
-  return destArray;
+  return dest;
 };
 
-export const cloneArray = src => src.slice();
+export const cloneArray = src => src && src.slice();
 
 export const copyJSON = src => JSON.parse(JSON.stringify(src));
 
 export const cloneJSON = src => JSON.parse(JSON.stringify(src));
 
-export const copyCopyable = (src, dest) => dest.copy(src);
+export const copyCopyable = (src, dest) => {
+  if (!src) {
```

Comment:
what is the logic here? do you mean `return dst;`?

Reply:
I think this is correct. You are copying from source (`src`) to destination (`dest`).

If `src === null` or `src === undefined` then we want `dest` to be `null` or `undefined`.

If `dest === null` or `dest === undefined` then we want `dest` to be a clone of `src`. We should keep this in mind because that means there could be a performance impact when defaulting a cloneable prop to null or undefined. Or you could manually use an object pool. Perhaps we can make a pooled prop type?

Also, notice I have the same behavior implemented for arrays.

Reply:
Okok, it makes sense. I'll just try to add some tests later so it could be useful as documentation too